### ISSUE TITLE
Error decoding Rotation created with pitch, yaw and roll

### ISSUE THREAD

**Issue by CaptainRedmuff** [nicklockwood/Euclid#42]
I am creating rotations using the `pitch: yaw: roll:` initialiser as below but when decoding these an error is thrown when checking the `isRotationMatrix` property during initialisation.

The following test recreates the failing use case:

```
func testEncodeAndDecodingRotationPitchYawRoll() throws {
    let rotation = Rotation(pitch: Angle(degrees: 10), yaw: Angle(degrees: 20), roll: Angle(degrees: 30))
    let encoded = try encode(rotation)
    XCTAssert(try rotation.isEqual(to: decode(encoded)))
}
```

**Comment by nicklockwood** [comment]
This seems to not be an encoding problem, but rather a bug in the `isRotationMatrix` check. I'm not sure what's happening there - I'll investigate further.

**Comment by nicklockwood** [comment]
@CaptainRedmuff fixed in 0.4.2
